modules/auxiliary/scanner/http/wp_perfect_survey_sqli.rb

```diff
@@ -0,0 +1,87 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'WordPress Plugin Perfect Survey 1.5.1 SQLi (Unauthenticated)',
+        'Description' => %q{
+          This module exploits a SQL injection vulnerability in the Perfect Survey
+          plugin for WordPress (version 1.5.1). An unauthenticated attacker can
+          exploit the SQLi to retrieve sensitive information such as usernames
+          and password hashes from the `wp_users` table.
+        },
+        'Author' => [
+          'Aaryan Golatkar', # Metasploit Module Creator
+          'Ron Jost'         # Vulnerability discovery
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['EDB', '50766'],
+          ['CVE', '2021-24762']
+        ],
+        'DisclosureDate' => '2021-10-05',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path to the WordPress installation', '/']),
+      OptBool.new('SHOW_FULL_RESPONSE', [false, 'Show the entire JSON response if username and password hash are not extracted', false]),
+      Opt::RPORT(80) # Default port for HTTP
+    ])
+  end
+
```

Comment:
The check method is missing, you can learn more [here](https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html).